### PR TITLE
Cleanup qemu-user, gdb<12, vmmap

### PR DESCRIPTION
**Pull request by patryk4815**
Tickets partially:
- #2636 
- #2637

BEFORE:
qemu-user 8.1:
image
qemu-user 6.2:
image

AFTER:
ubuntu24.04 - qemu-user 8.1:
image

ubuntu 22.04 - qemu-user 6.2:
image

### CODE REVIEW

File: pwndbg/gdblib/vmmap.py [hunk 1]
```diff
@@ -535,16 +527,17 @@ def proc_tid_maps() -> Tuple[pwndbg.lib.memory.Page, ...] | None:
 
     tid = pwndbg.aglib.proc.tid
     locations = [
+        # Linux distro
         f"/proc/{tid}/maps",
-        f"/proc/{tid}/map",
```
Comment (patryk4815) [inline pwndbg/gdblib/vmmap.py:539]: I couldn't find it anywhere

File: pwndbg/aglib/file.py [hunk 2]
```diff
@@ -43,6 +45,29 @@ def get_proc_exe_file() -> str:
...
+    if not pwndbg.aglib.remote.is_remote():
+        return False
+    elif pwndbg.aglib.qemu.is_qemu_kernel():
+        return False
+
+    # Some[1] gdb servers don't implement vFile packets.
+    # [1] - qemu-user <8.1
+    # [1] - Rosetta2
+    # [1] - maybe embedded probe like: `Black Magic Probe V2.3`
+    # WTF: There is no indication in `qSupported` when `vFile` packets are supported
```
Comment (disconnect3d) [inline pwndbg/aglib/file.py:59]: Does qemu-user>=8.1 do indicate this?

Is this maybe a bug in Rosetta2 and embedded probes?

File: pwndbg/aglib/file.py [hunk 3]
```diff
@@ -107,72 +134,138 @@ def get(path: str) -> bytes:
...
+    """
+    Reads the entire content of a file on the remote system.
+
+    :param filename: The path to the file (string).
+    :param chunk_size: The number of bytes to read in each iteration (integer).
+    :return: The complete content of the file as bytes.
+    """
+    fd = None
+    try:
+        # 0 = readonly
```
Comment (disconnect3d) [inline pwndbg/aglib/file.py:207]: This is comment about file mode or something else?

**Review by disconnect3d — APPROVED** [review]

File: pwndbg/aglib/file.py (hunk 2 above)
Comment (patryk4815) [reply pwndbg/aglib/file.py:59]: `qemu-user>=8.1` sends only `qXfer:exec-file:read` but what this mean at all? 
https://sourceware.org/legacy-ml/gdb-testers/2015-q2/msg02348.html

**Pull request merged**